### ISSUE TITLE
🐛 useExhaustiveDependencies rule false positive

### ISSUE THREAD

**Issue by leevare** [biomejs/biome#8967]
### Environment information

<details>

```bash

CLI:
  Version:                      2.3.14
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v20.19.3
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/10.8.2

Biome Configuration:
  Error:                        The configuration schema version does not match the CLI version 2.3.14
  Status:                       Loaded with errors
  Path:                         biome.jsonc
  Formatter enabled:            false
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0

```

</details>

### Rule name

`useExhaustiveDepdendencies`

### Playground link

https://biomejs.dev/playground/?lintRules=useExhaustiveDependencies&code=aQBtAHAAbwByAHQAIAB7AHUAcwBlAEUAZgBmAGUAYwB0AH0AIABmAHIAbwBtACAAJwByAGUAYQBjAHQAJwAKAAoAZgB1AG4AYwB0AGkAbwBuACAAQwBvAG0AcABvAG4AZQBuAHQAKABwAHIAbwBwAHMAKQAgAHsACgAgACAAYwBvAG4AcwB0ACAAewBkAGEAdABhACwAIAAuAC4ALgByAGUAcwB0AFAAcgBvAHAAcwB9ACAAPQAgAHAAcgBvAHAAcwAKACAAIABjAG8AbgBzAHQAIAB7AHAAcgBvAHAAMQB9ACAAPQAgAHIAZQBzAHQAUAByAG8AcABzAAoACgAgACAAdQBzAGUARQBmAGYAZQBjAHQAKAAoACkAIAA9AD4AIAB7AAoAIAAgACAAIABjAG8AbgBzAG8AbABlAC4AbABvAGcAKABwAHIAbwBwADEAKQAKACAAIAB9ACwAIABbAHAAcgBvAHAAMQBdACkACgAKACAAIAB1AHMAZQBFAGYAZgBlAGMAdAAoACgAKQAgAD0APgAgAHsACgAgACAAIAAgAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoAHIAZQBzAHQAUAByAG8AcABzAC4AcAByAG8AcAAxACkACgAgACAAfQAsACAAWwByAGUAcwB0AFAAcgBvAHAAcwAuAHAAcgBvAHAAMQBdACkACgAKACAAIAByAGUAdAB1AHIAbgAgAG4AdQBsAGwACgB9AA%3D%3D&ruleDomains.react=recommended

### Expected result

There should be no errors.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by mdevils** [comment]
The fix will be released soon.
